finetuner/experiment.py

```diff
@@ -216,40 +210,60 @@ def _create_config_for_run(
         """
         callbacks = kwargs[CALLBACKS] if kwargs.get(CALLBACKS) else []
         callbacks = [
-            {
-                NAME: callback.__class__.__name__,
-                OPTIONS: {
+            config.CallbackConfig(
+                name=callback.__class__.__name__,
+                options={
                     field.name: getattr(callback, field.name)
                     for field in fields(callback)
                 },
-            }
+            )
             for callback in callbacks
         ]
-        return {
-            MODEL: {
-                NAME: model,
-                FREEZE: kwargs.get(FREEZE),
-                OUTPUT_DIM: kwargs.get(OUTPUT_DIM),
-                OPTIONS: kwargs.get(MODEL_OPTIONS) or {},
-                ONNX: kwargs.get(ONNX) or False,
-            },
-            DATA: {
-                TRAIN_DATA: train_data,
-                EVAL_DATA: kwargs.get(EVAL_DATA),
-                NUM_WORKERS: kwargs.get(NUM_WORKERS),
-            },
-            CALLBACKS: callbacks,
-            HYPER_PARAMETERS: {
-                LOSS: kwargs.get(LOSS),
-                OPTIMIZER: kwargs.get(OPTIMIZER),
-                OPTIMIZER_OPTIONS: kwargs.get(OPTIMIZER_OPTIONS) or {},
-                MINER: kwargs.get(MINER),
-                MINER_OPTIONS: kwargs.get(MINER_OPTIONS) or {},
-                BATCH_SIZE: kwargs.get(BATCH_SIZE),
-                LEARNING_RATE: kwargs.get(LEARNING_RATE),
-                EPOCHS: kwargs.get(EPOCHS),
-                SCHEDULER_STEP: kwargs.get(SCHEDULER_STEP),
-            },
-            EXPERIMENT_NAME: experiment_name,
-            RUN_NAME: run_name,
-        }
+        model = config.ModelConfig(
+            name=model,
+            output_dim=kwargs.get(OUTPUT_DIM),
+        )
+        if kwargs.get(FREEZE):
+            model.freeze = kwargs.get(FREEZE)
```

Comment:
so this is the same as?
```suggestion
        model.freeze = kwargs.get(FREEZE, False)
```

Reply:
also applies to other get

Reply:
make sense